### PR TITLE
Add option to substitute `isWeb` and `shouldBeUseWeb` in babel plugin.

### PR DESCRIPTION
**Pull request by bartlomiejbloniarz**
## Summary

This PR changes the Babel plugin to substitute `isWeb()` and `shouldBeUseWeb()` calls with `true` so that Webpack is able to recognize dead code.

## Test plan
Modify the NextExample/babel.config.js file:
```js
['react-native-reanimated/plugin', { omitNativeOnlyData: true, substituteWebPlatformChecks: true }]
```
Run yarn build in NextExample and verify that the difference between reanimated and noreanimated is around 34KB.

### CODE REVIEW

**Review by tomekzaw — COMMENTED** [review]
I have a crazy idea, why don't we just create a new file `PlatformChecker.web.ts` with the following contents? 😄 

```ts
function isWeb() {
  return true;
}

function shouldBeUseWeb() {
  return true;
}

// ...
```

File: plugin/src/plugin.ts [hunk 1]
```diff
@@ -27,7 +28,12 @@ module.exports = function (): PluginItem {
     visitor: {
       CallExpression: {
         enter(path: NodePath<CallExpression>, state: ReanimatedPluginPass) {
-          runWithTaggedExceptions(() => processForCalleesWorklets(path, state));
+          runWithTaggedExceptions(() => {
+            processForCalleesWorklets(path, state);
+            if (state.opts.substituteWebPlatformChecks) {
+              substituteWebCallExpression(path);
+            }
+          });
```
Comment (tjzel) [inline plugin/src/plugin.ts:36]: We have to keep in mind this won't transform `shouldBeUseWeb` or `isWeb` inside worklets. While it's not a valid case to call them there, we should at least have it included in tests.

File: plugin/src/types.ts
Comment (tjzel) [inline plugin/src/types.ts:1]: I don't quite remember, didn't we agree to use some better names here? Or was this regarding some other PR.

**Review by tjzel — CHANGES_REQUESTED** [review]

File: plugin/src/types.ts
Comment (bartlomiejbloniarz) [reply plugin/src/types.ts:1]: We agreed on `omitNativeOnlyData` in a separate pr. I just need to merge those changes here.

**Review by tjzel — APPROVED** [review]
Me likes 🌵

**Pull request merged**